File: corpus/cs1/0013-nagol/solutions/a-optimal/solution.cpp
#include <iostream>

using namespace std;

int main() {
    long long l, c, x, y;
    while (cin >> l >> c >> x >> y) {
        long long index = x * c + y;
        if (index % 2 == 0) {
            cout << "Direita" << endl;
        } else {
            cout << "Esquerda" << endl;
        }
    }
    return 0;
}

File: corpus/cs1/0013-nagol/solutions/a-optimal/Solution.java
import java.util.Scanner;

public class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        while (sc.hasNextLong()) {
            long l = sc.nextLong();
            long c = sc.nextLong();
            long x = sc.nextLong();
            long y = sc.nextLong();
            
            long index = x * c + y;
            if (index % 2 == 0) {
                System.out.println("Direita");
            } else {
                System.out.println("Esquerda");
            }
        }
        sc.close();
    }
}
